sdk/ml/azure-ai-ml/azure/ai/ml/entities/_schedule/schedule.py

```diff
@@ -78,6 +79,19 @@ def _resolve_cls_and_type(cls, data, params_override):  # pylint: disable=unused
             return MonitorSchedule, None
         return JobSchedule, None
 
+    @classmethod
+    def _from_rest_object(cls, obj: RestSchedule) -> "Schedule":
+
+        if obj.properties.action.action_type == RestScheduleActionType.CREATE_JOB:
+            return JobSchedule._from_rest_object(obj)
+        msg = f"Unsupported schedule type {obj.properties.action.action_type}"
+        raise ScheduleException(
+            message=msg,
+            no_personal_data_message=msg,
+            target=ErrorTarget.SCHEDULE,
+            error_category=ErrorCategory.SYSTEM_ERROR,
+        )
+
     @property
     def create_job(self) -> None:
         module_logger.warning("create_job is not a valid property of %s", str(type(self)))
```

Comment:
Can we add a `return None` here, just to be explicit?

Reply:
added